pkg/tracing/provider/jaeger/config_yaml.go

```diff
@@ -0,0 +1,213 @@
+package jaeger
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"github.com/uber/jaeger-client-go"
+	"github.com/uber/jaeger-client-go/config"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	ServiceName            string `yaml:"service_name"`
+	Disabled               string `yaml:"disabled"`
+	RPCMetrics             string `yaml:"rpc_metrics"`
+	Tags                   string `yaml:"tags"`
+	SamplerType            string `yaml:"sampler_type"`
+	SamplerParam           string `yaml:"sampler_param"`
+	SamplerManagerHostPort string `yaml:"sampler_manager_host_port"`
+	SamplerMaxOperations   string `yaml:"sampler_max_operations"`
+	SamplerRefreshInterval string `yaml:"sampler_refresh_interval"`
+	ReporterMaxQueueSize   string `yaml:"reporter_max_queue_size"`
+	ReporterFlushInterval  string `yaml:"reporter_flush_interval"`
+	ReporterLogSpans       string `yaml:"reporter_log_spans"`
+	Endpoint               string `yaml:"endpoint"`
+	User                   string `yaml:"user"`
+	Password               string `yaml:"password"`
+	AgentHost              string `yaml:"agent_host"`
+	AgentPort              string `yaml:"agent_port"`
+}
+
+// FromYaml uses config YAML to set the tracer's Configuration
+func FromYaml(cfg []byte) (*config.Configuration, error) {
+	conf := &Config{}
+
+	if err := yaml.Unmarshal(cfg, &conf); err != nil {
+		return nil, err
+	}
+
+	c := &config.Configuration{}
+
+	if e := conf.ServiceName; e != "" {
+		c.ServiceName = e
+	}
+
+	if e := conf.RPCMetrics; e != "" {
+		if value, err := strconv.ParseBool(e); err == nil {
+			c.RPCMetrics = value
+		} else {
+			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", conf.RPCMetrics, e)
+		}
+	}
+
+	if e := os.Getenv(conf.Disabled); e != "" {
+		if value, err := strconv.ParseBool(e); err == nil {
+			c.Disabled = value
+		} else {
+			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", conf.Disabled, e)
+		}
+	}
+
+	if e := conf.Tags; e != "" {
```

Comment:
Why `e`? (: 
Why this is better than just:
```
if conf.Tags != "" {
  c.Tags = confg.Tags
}
```

Just curious (:

Reply:
It's copy/paste of https://github.com/jaegertracing/jaeger-client-go/blob/master/config/config_env.go )